dali/operators/reader/loader/coco_loader.cc

```diff
@@ -219,19 +263,41 @@ void ParseAnnotations(LookaheadParser &parser, std::vector<Annotation> &annotati
         if (parse_rle && parser.PeekType() == kObjectType) {
           annotation.tag_ = Annotation::RLE;
           parser.EnterObject();
+          rle_str.clear();
+          rle_uints.clear();
+          int h = -1, w = -1;
           while (const char* another_key = parser.NextObjectKey()) {
             if (0 == std::strcmp(another_key, "size")) {
               RAPIDJSON_ASSERT(parser.PeekType() == kArrayType);
               parser.EnterArray();
               parser.NextArrayValue();
-              annotation.rle_.h_ = parser.GetInt();
+              h = parser.GetInt();
               parser.NextArrayValue();
-              annotation.rle_.w_ = parser.GetInt();
+              w = parser.GetInt();
               parser.NextArrayValue();
             } else if (0 == std::strcmp(another_key, "counts")) {
-              annotation.rle_.rle_ = parser.GetString();
+              if (parser.PeekType() == kStringType) {
+                rle_str = parser.GetString();
+              } else if (parser.PeekType() == kArrayType) {
+                parser.EnterArray();
+                while (parser.NextArrayValue()) {
+                  rle_uints.push_back(parser.GetInt());
+                }
+              } else {
+                parser.SkipValue();
+              }
+            } else {
+              parser.SkipValue();
             }
           }
+          DALI_ENFORCE(h > 0 && w > 0, "Invalid or missing mask sizes");
+          if (!rle_str.empty()) {
+            annotation.rle_ = RLEMask(h, w, rle_str.c_str());
```

Comment:
Can we do any sanity check for `h`, `w` and `rle_str` size?

Reply:
What do you mean, there's an enforce for h and w in line 293, and I am only parsing if the string is not empty. There's no info about the mask dimensions in the string itself.